app/src/features/incentivization/components/CreditsProgressbar/CreditsProgessbar.tsx

```diff
@@ -10,23 +10,78 @@ import {
   getIsIncentivizationDetailsLoading,
 } from "store/features/incentivization/selectors";
 import { getTotalCredits } from "features/incentivization/utils";
+import { getUserAuthDetails } from "store/selectors";
+import { getAvailableBillingTeams } from "store/features/billing/selectors";
 import "./creditsProgessbar.scss";
 
-// TODO: fix isInModal in redeem flow
-export const CreditsProgressBar: React.FC<{ isInModal?: boolean }> = ({ isInModal = true }) => {
+interface CreditsProgressBarProps {
+  source: string;
+}
+
+export const CreditsProgressBar: React.FC<CreditsProgressBarProps> = ({ source }) => {
   const [isRedeemModalVisible, setIsRedeemModalVisible] = useState(false);
 
+  const user = useSelector(getUserAuthDetails);
   const isLoading = useSelector(getIsIncentivizationDetailsLoading);
   const milestones = useSelector(getIncentivizationMilestones);
   const userMilestoneDetails = useSelector(getIncentivizationUserMilestoneDetails);
+  const billingTeams = useSelector(getAvailableBillingTeams);
+
+  const isUserHasActiveSubscription = user?.details?.planDetails?.status === "active";
+  const isUserInBillingTeam =
+    billingTeams?.length && billingTeams?.some((team) => user?.details?.profile?.uid in team.members);
```

Comment:
should we memoize this? This will be iterating on all the available billing teams for all re-renders